content/posts/why-meson/index.md

```diff
@@ -0,0 +1,248 @@
+not `/usr`, we have chosen to keep RPATHs to make the installation process as
+simple as possible, without requiring users to mess with their environment to
+make sure the binaries can find the proper libraries. Packagers, who usually
+use `/usr` as prefix, should not be affected by this decision, but they can
+anyway disable it by specifying `-Dlocal=false` when running meson.
+
+## Reviewing a PR and testing changes
+
+When testing a PR with a fix or comparing multiple changes, you need to have
+access to multiple versions of Rizin. Doing this with ACR/Make is of course
+possible, but it usually involves installing everything in separated
+directories and making sure your environment variables (e.g. PATH,
+LD_LIBRARY_PATH, etc.) are correctly set. With meson you can build one
+version (e.g. from `dev` branch) with `meson --prefix=~/.local build-dev;
+ninja -C build-dev`, then switch branch with `git checkout my-other-branch`
+and build Rizin again with `meson --prefix=~/.local build-pr; ninja -C
+build-pr`. Due to the RPATH used by default, as mentioned above, each build
+directory can be used without installation to actually run the Rizin tools.
+At that point you can quickly compare the results of
+`./build-dev/binrz/rizin/rizin` and `./build-pr/binrz/rizin/rizin`.
+
+# Conclusion
+
+Of course it's not all perfect with meson either. Right now the meson build
+system is missing some features that were only available with ACR/Make.
+
+To uninstall Rizin you have to run `ninja -C build uninstall` from the same
+build directory you used to run the `install` step, otherwise it will not
+uninstall files. However if during `install` step we add any custom
+installation script (e.g. to sign your rizin binary in macOS), there is no
+counter part to actually have an uninstall script. That said, nothing
+prevents us from having a custom target similar to what ACR/Makefile system
+does to manually remove, with a script, the installed files, but we believe
+proper file tracking should be done by distributions and packages.
+
+Meson is quite new and, although rare, you may find issues from time to time.
+That said, its community is healthy and active so you can count on them to
+fix these problems as soon as possible or provide help, also thanks to the
+many big projects that have switched to meson in the last years.
+
+All in all, we hope to make it easier for us and for developers/users to
+build Rizin. We are trying to build a good Reverse Engineering Framework and
+we want to focus our efforts on this rather than dealing with the limitations
+of a niche build system.
+
+We hope this was a useful overview for developers to understand why we
+switched to meson. If you find issues or find particular installation setups
+difficult or missing, feel free to open a bug in GitHub and we will be happy
+to either guide you through a solution or develop the fix according to our
+roadmap.
```

Comment:
After reaching the end, one question could be "why not CMake"?
I don't remember the exact reason for deciding to take a look at meson rather than CMake...
If you remember, it might be worth mentioning it somewhere in the blogpost, saying that it was considered but not chosen over `meson` although it is much more popular for whatever reason.

Reply:
I would like I think I would put at https://github.com/rizinorg/website/pull/11/files#diff-ed8aa5f0beac71a2420fe46ca51f8a8718c111a9b47ff2740cdd48a35bb0a84eR57 this:
```
Other very valid alternatives such as CMake were considered, however we
preferred to keep working with Meson, which was already tested and tried for
a long time, rather than starting completely from scratch with yet another
build system.
```

Which is essentially the reason, though not a big one, but it is what it is. We wanted to switch to a more "high level" build system and cmake and meson were our options, but with meson we already had a lot of things done. @ITAYC0HEN

Reply:
I'd omit "yet another" and would clarify that "tested" with Rizin and not just "was already tested"